src/client.rs

```diff
@@ -0,0 +1,129 @@
+#![deny(missing_docs)]
+use crate::error::NysmError;
+use async_trait::async_trait;
+use std::collections::VecDeque;
+
+/// Represents a response from a secret provider that wraps around
+/// the actual secret value held by a secret.
+#[derive(Default)]
+pub struct GetSecretValueResult {
+  /// Secret string value associated with Secret.
+  pub secret: String,
+}
```

Comment:
What is this string value suppose to be exactly? A serialized value I'd assume?

Reply:
It's the actual data associated with a particular secret. I wouldn't call it serialized per se.